dts/bindings/gpio/nxp,lpc-gpio.yaml

```diff
@@ -28,6 +28,20 @@ properties:
       - 6
       - 7
 
+  int-source:
+    type: string
+    default: "pint"
```

Comment:
I would prefer to leave pint as the default. It seems all the LPC devices use it as an interrupt source (which is why this driver was using PINT to manage interrupts previously). For the RT devices, if PINT is the default then interrupt installation will simply fail on GPIO ports that cannot use PINT as a source, until they are configured otherwise.

Reply:
Considering that PINT is not part of the GPIO IP and it is a separate IP block only present on some SOCs, it does not feel right to say the default hardware configuration for this GPIO IP block uses PINT. I would even go so far as to say that PINT should have it's own DT node. But since it is based on using the GPIO pins, it might make sense to make it a child node of the GPIO or vice versa, or to make a phandle property for pint. I just think it needs to be clear in devicetree somehow that PINT is not part of the LPC GPIO IP block. And I think PINT could have an interrupt_controller driver similar to how STM32 has an [EXTI driver](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/interrupt_controller/intc_exti_stm32.c) for dealing with external interrupt sources, which seems to be a very similar concept to what PINT is.

Reply:
> And I think PINT could have an interrupt_controller driver similar to how STM32 has an [EXTI driver](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/interrupt_controller/intc_exti_stm32.c) for dealing with external interrupt sources, which seems to be a very similar concept to what PINT is.

I agree. The PINT needs to be represented in DTS. Comparing it to the STM32 EXTI seems accurate.